File: C++/longest-unique-substring.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string s;
    getline(cin, s);

    unordered_map<char, int> lastSeen;
    int maxLen = 0, start = 0;

    for (int i = 0; i < (int)s.size(); i++) {
        char ch = s[i];
        if (lastSeen.count(ch) && lastSeen[ch] >= start) {
            start = lastSeen[ch] + 1;
        }
        lastSeen[ch] = i;
        maxLen = max(maxLen, i - start + 1);
    }

    cout << maxLen << endl;
    return 0;
}

File: JS/longest-unique-substring.js
function longestUniqueSubstring(str) {
    const lastSeen = new Map();
    let maxLen = 0;
    let start = 0;

    for (let i = 0; i < str.length; i++) {
        const ch = str[i];
        if (lastSeen.has(ch) && lastSeen.get(ch) >= start) {
            start = lastSeen.get(ch) + 1;
        }
        lastSeen.set(ch, i);
        maxLen = Math.max(maxLen, i - start + 1);
    }

    return maxLen;
}

const input = require('fs').readFileSync('/dev/stdin', 'utf8').trim();
console.log(longestUniqueSubstring(input));